server/src/main/java/io/druid/segment/loading/SegmentLoaderConfig.java

```diff
@@ -85,7 +85,10 @@ public int getNumBootstrapThreads()
   public File getInfoDir()
   {
     if (infoDir == null) {
-      infoDir = new File(locations.get(0).getPath(), "info_dir");
+      StorageLocationConfig storageLocation = locations.get(0);
+      if (storageLocation != null) {
+        infoDir = new File(storageLocation.getPath(), "info_dir");
+      }
```

Comment:
This method will throw an IndexOutOfBoundsException if locations is empty (due to the `.get(0)`) and an NPE at the call site if locations is non-empty, but has a null value. So I suggest detecting these conditions and throwing a clearly worded error message like "You have no segment cache locations defined. Please configure druid.segmentCache.locations to use one or more locations."